src/peep/peep.c

```diff
@@ -315,7 +315,8 @@ static void peep_leave_park(rct_peep* peep){
 		peep->peep_flags &= ~PEEP_FLAGS_PARK_ENTRANCE_CHOSEN;
 	}
 
-	peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_GO_HOME, 0xFF);
+	if (peep->peep_flags & PEEP_FLAGS_TRACKING) { return; }
+	else { peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_GO_HOME, 0xFF); }
```

Comment:
Because you have blocked it from reaching `peep_leave_park` if the peep is tracked, you don't need it here as well.